deepchecks/core/display.py

```diff
@@ -0,0 +1,368 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+# pylint: disable=unused-argument
+"""Module with check/suite result display strategy in different envs."""
+import abc
+import html
+import io
+import sys
+import typing as t
+
+import plotly.io as pio
+from IPython.core.display import display, display_html
+from ipywidgets import Widget
+
+from deepchecks.core.serialization.abc import HTMLFormatter, HtmlSerializer, IPythonSerializer, WidgetSerializer
+from deepchecks.utils.ipython import is_colab_env
+from deepchecks.utils.strings import create_new_file_name, get_random_string, widget_to_html, widget_to_html_string
+
+if t.TYPE_CHECKING:
+    from wandb.sdk.data_types.base_types.wb_value import WBValue  # pylint: disable=unused-import
+
+__all__ = ['DisplayableResult', 'save_as_html', 'display_in_gui']
+
+
+T = t.TypeVar('T')
+
+
+class DisplayableResult(abc.ABC):
+    """Display API for the check/suite result objects."""
+
+    @property
+    @abc.abstractmethod
+    def widget_serializer(self) -> WidgetSerializer[t.Any]:
+        """Return WidgetSerializer instance."""
+        raise NotImplementedError()
+
+    @property
+    @abc.abstractmethod
+    def ipython_serializer(self) -> IPythonSerializer[t.Any]:
+        """Return IPythonSerializer instance."""
+        raise NotImplementedError()
+
+    @property
+    @abc.abstractmethod
+    def html_serializer(self) -> HtmlSerializer[t.Any]:
+        """Return HtmlSerializer instance."""
+        raise NotImplementedError()
+
+    def show(
+        self,
+        as_widget: bool = True,
```

Comment:
If as_widgets is false, doesn't it mean we need to use the HTML serializer?

Reply:
not in the case of the `show` method 
here we are using `IPythonSerializer`

take a look at how it deals with plotly figures 
generally saying, when we use `IPythonSerializer` 
we delegate display responsibility to the plotly figure itself

but in the case of `show_in_iframe` method we `HtmlSerializer`

Reply:
So If we got the IFrame I wonder if it's smart to continue support also the IPython serializer

Reply:
nope, we need it, I mean `IPythonSerializer`

shortly:
- we cannot use `HtmlSerializer` in the case of `show(as_widget=False)`, because we do not know how plotly figures should be serialized in the current context (therefore we use `IPythonSerializer` and  delegate this task to the plotly itself)
- we cannot use `IPythonSerializer` in the case of `show_in_iframe(as_widget=False) / show_in_window(as_widget=False)`, becasue we cannot use `IPythonSerializer` to produce html document

Reply:
yea I meant we will just drop the as_widgets, we will always do it as widgets, unless it's non-interactive